File: classic puzzle-medium/gravity.py
#https://www.codingame.com/training/medium/gravity
#GRAVITY
w,h=[int(i) for i in input().split()]
cnt=[0]*w #count the number of tiles per column
for i in range(h):
    line=input()
    for j in range(len(line)):
        if line[j]=="#":
            cnt[j]+=1
# #'s numbers per column if it above H_i than adding "#"
#H_i : numbers at that hieght
for h_i in range(h,0,-1):
    print("".join(["#" if cnt[j]>=h_i else "." for j in range(w)]))

File: classic puzzle-medium&above/gravity.rb
#https://www.codingame.com/training/medium/gravity
#GRAVITY
w,h=gets.split.map &:to_i
cnt=[0]*w #count the number of tiles per column
h.times{
  s=gets.chomp
  (0...s.size).map{cnt[_1]+=1 if s[_1]=="#"}
}
# #'s numbers per column if it above H_i than adding "#"
#H_i : numbers at that hieght
(h...0).step(-1).map{|hieght|
    puts (0...w).map{|l|cnt[l]>=hieght ? "#":"."}*""
}
